terraform/aws/main.tf

```diff
@@ -0,0 +1,551 @@
+  name       = "gpu-exporter"
+  repository = "https://nvidia.github.io/dcgm-exporter/helm-charts"
+  chart      = "dcgm-exporter"
+  namespace  = "kube-system"
+}
+
+
+resource "helm_release" "karpenter" {
+  namespace        = "karpenter"
+  create_namespace = true
+
+  name       = "karpenter"
+  repository = "https://charts.karpenter.sh"
+  chart      = "karpenter"
+  version    = "v0.10.0"
+
+  set {
+    name  = "serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn"
+    value = module.karpenter_irsa.iam_role_arn
+  }
+
+  set {
+    name  = "clusterName"
+    value = module.eks.cluster_id
+  }
+
+  set {
+    name  = "clusterEndpoint"
+    value = module.eks.cluster_endpoint
+  }
+
+  set {
+    name  = "aws.defaultInstanceProfile"
+    value = aws_iam_instance_profile.karpenter.name
+  }
+}
+resource "null_resource" "patch_aws_cni" {
+  depends_on = [
+     module.eks.node_group_id,
+     helm_release.argo
+  ]
+
+  provisioner "local-exec" {
+    command = <<EOF
+# do all those commands to get kubectl and auth info, then run:
+kubectl set env daemonset aws-node -n kube-system ENABLE_PREFIX_DELEGATION=true WARM_PREFIX_TARGET=2
+EOF
+  }
+}
+
+resource "helm_release" "ingress-nginx" {
+  count      = var.enable-ingress-nginx && !var.enable-grafana ? 1 : 0
+
+  name       = "ingress-nginx"
+  chart      = "../../charts/ingress-nginx"
+  namespace  = "ingress-nginx"
+  create_namespace = true
+
+  values = [
+    "${file("../../charts/ingress-nginx/values.yaml")}"
+  ]
+}
+
+resource "helm_release" "ingress-nginx-grafana" {
+  count      =  var.enable-ingress-nginx  && var.enable-grafana ? 1 : 0
+  name       = "ingress-nginx"
+  chart      = "../../charts/ingress-nginx"
+  namespace  = "ingress-nginx"
+  create_namespace = true
+
+  values = [
+    "${file("../../charts/values-overrides/internal/eks-dev-values.yaml")}"
+  ]
+}
+resource "aws_security_group" "remote_access" {
+  name_prefix = "${local.cluster_name}-remote-access"
+  description = "Allow remote SSH access"
+  vpc_id      = module.vpc.vpc_id
+
+  ingress {
+    description = "SSH access"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["10.0.0.0/8"]
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+        # This will tag the launch template created for use by Karpenter
+        "karpenter.sh/discovery/${local.cluster_name}" = local.cluster_name
+      }
+}
+
+resource "helm_release" "grafana-stack" {
+  depends_on = [
+        kubectl_manifest.argo_apps
+    ]
+  count      = var.enable-grafana ? 1 : 0
+  name       = "prometheus"
+  chart      = "../../charts/kube-prometheus-stack"
+  namespace  = "prometheus"
+  create_namespace = true
+
+  values = [
+    "${file("../../charts/values-overrides/internal/eks-dev-values.yaml")}"
+  ]
+}
+
+resource "helm_release" "morpheus-ai-engine" {
```

Comment:
strange for morpheus to be here already?

Reply:
It's not in here twice.. one is deploying the mlflow plug-in and one is deploying morpheus